RELEASE_NOTES.md

```diff
@@ -2,6 +2,20 @@
 
 ## Unreleased
 
+## 3.0.0 (2016-02-07)
+
+#### BREAKING CHANGES
+* `git hack`: no longer accepts a parent branch
```

Comment:
I looked through the issues before making this PR and there wasn't any discussion for this in particular. There is an issue for [git prepend](https://github.com/Originate/git-town/issues/766) but `git append` and the change to `git hack` I think just came out of the PRs. @kevgo were you working off any particular issues for those changes?